pkg/cmd/version/version.go

```diff
@@ -18,21 +18,109 @@ package version
 
 import (
 	"fmt"
+	"time"
 
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
 
+	v1 "github.com/heptio/ark/pkg/apis/ark/v1"
+	"github.com/heptio/ark/pkg/builder"
 	"github.com/heptio/ark/pkg/buildinfo"
+	"github.com/heptio/ark/pkg/client"
+	"github.com/heptio/ark/pkg/cmd"
+	arkv1client "github.com/heptio/ark/pkg/generated/clientset/versioned/typed/ark/v1"
 )
 
-func NewCommand() *cobra.Command {
+func NewCommand(f client.Factory) *cobra.Command {
+	timeout := time.Minute
+	clientOnly := false
+
 	c := &cobra.Command{
 		Use:   "version",
 		Short: "Print the ark version and associated image",
-		Run: func(cmd *cobra.Command, args []string) {
-			fmt.Printf("Version: %s\n", buildinfo.Version)
-			fmt.Printf("Git commit: %s\n", buildinfo.FormattedGitSHA())
+		Run: func(c *cobra.Command, args []string) {
+			fmt.Println("Client:")
+			fmt.Printf("\tVersion: %s\n", buildinfo.Version)
+			fmt.Printf("\tGit commit: %s\n", buildinfo.FormattedGitSHA())
+
+			if clientOnly {
+				return
+			}
+
+			client, err := f.Client()
+			cmd.CheckError(err)
+
+			serverStatus, err := getServerStatus(client.ArkV1(), f.Namespace(), timeout)
+			if err != nil {
+				fmt.Printf("<error getting server version: %s>\n", err)
+				return
+			}
+
+			fmt.Println("Server:")
+			fmt.Printf("\tVersion: %s\n", serverStatus.Status.ServerVersion)
 		},
 	}
 
+	c.Flags().DurationVar(&timeout, "timeout", timeout, "maximum time to wait for server version to be reported")
+	c.Flags().BoolVar(&clientOnly, "client-only", clientOnly, "only get ark client version, not server version")
+
 	return c
 }
+
+func getServerStatus(client arkv1client.StatusRequestsGetter, namespace string, timeout time.Duration) (*v1.StatusRequest, error) {
+	req := builder.NewStatusRequest().Namespace(namespace).GenerateName("ark-cli-").StatusRequest()
+
+	created, err := client.StatusRequests(namespace).Create(req)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	defer client.StatusRequests(namespace).Delete(created.Name, nil)
+
+	listOptions := metav1.ListOptions{
```

Comment:
We have this same logic in `downloadrequest.Stream()`. It might be nice in the future to extract this to a helper, with a custom handler function to call in the `watch.Modified` case.